monai/handlers/metrics_saver.py

```diff
@@ -0,0 +1,176 @@
+        summary_ops: Optional[Union[str, Sequence[str]]] = None,
+        save_rank: int = 0,
+        delimiter: str = "\t",
+    ) -> None:
+        self.save_dir = save_dir
+        self.metrics = ensure_tuple(metrics) if metrics is not None else None
+        self.metric_details = ensure_tuple(metric_details) if metric_details is not None else None
+        self.batch_transform = batch_transform
+        self.summary_ops = ensure_tuple(summary_ops) if summary_ops is not None else None
+        self.save_rank = save_rank
+        self.deli = delimiter
+        self._filenames: List[str] = []
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(Events.STARTED, self._started)
+        engine.add_event_handler(Events.ITERATION_COMPLETED, self._get_filenames)
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self)
+
+    def _started(self, engine: Engine) -> None:
+        self._filenames = []
+
+    def _get_filenames(self, engine: Engine) -> None:
+        if self.metric_details is not None:
+            _filenames = list(ensure_tuple(self.batch_transform(engine.state.batch)["filename_or_obj"]))
+            self._filenames += _filenames
+
+    def __call__(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        if not os.path.exists(self.save_dir):
+            os.makedirs(self.save_dir)
+
+        ws = idist.get_world_size()
+        if self.save_rank >= ws:
+            raise ValueError("target rank is greater than the distributed group size.s")
+
+        if self.metrics is not None and len(engine.state.metrics) > 0:
+            if idist.get_rank() == self.save_rank:
+                # only save metrics to file in specified rank
+                with open(os.path.join(self.save_dir, "metrics.csv"), "w") as f:
+                    for k, v in engine.state.metrics.items():
+                        if k in self.metrics or "*" in self.metrics:
+                            f.write(f"{k}{self.deli}{str(v)}\n")
+
+        if (
+            self.metric_details is not None
+            and hasattr(engine.state, "metric_details")
+            and len(engine.state.metric_details) > 0
+        ):
+            _filenames = self.deli.join(self._filenames)
+
+            if ws > 1:
+                if get_torch_version_tuple() > (1, 6, 0):
+                    # all gather across all processes
+                    _filenames = self.deli.join(idist.all_gather(_filenames))
+                else:
+                    raise RuntimeError(
+                        "MetricsSaver can not save metric details in distributed mode with PyTorch < 1.7.0."
+                    )
+            if idist.get_rank() == self.save_rank:
+                _files = _filenames.split(self.deli)
```

Comment:
Hi @Nic-Ma  sorry I fogot to mention it in the earlier comments: this is a nice utility from eval results to CSV files. could you please make it a standalone API? https://github.com/Project-MONAI/MONAI/issues/886 also mentions xlsx, html format, we need to think about extending to support these formats...

Reply:
Sure, I already extracted the `writing` logic into standalone APIs, I think it's easy to extend support for other formats later. Could you please help review it again?

Thanks.